### ISSUE TITLE
[SUGGESTION] Can't get code coverage when running tests via Test UI

### ISSUE THREAD

**Issue by julioct** [microsoft/vscode-dotnettools#554]
### Describe the feature you'd like

Getting code coverage involves passing additional parameters to dotnet test. Like this:

dotnet test <path to test project> --collect "XPlat Code Coverage" --settings <path to runsettings file>

I can't find a way to configure the C# extension or the Dev Kit to set those additional parameters.

Therefore any time I want to see if my coverage increased I can't rely on the UI provided by the Dev Kit. I have to find another way to run my tests with the required parameters.

### Alternatives considered

The only way to get code coverage is by setting a separate task in VS Code that runs dotnet test independently of the VS Code Test UI.

### Environment Information

OS: Windows 11
VS Code version: Code - Insiders 1.83.0-insider
Extension version: 2.2.10

**Comment by devklick** [comment]
I think it should be possible to use a [`.runsettings`](https://learn.microsoft.com/en-us/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file?view=vs-2022) with the relevant configuration to collect code coverage. As per the [linked issue](https://github.com/microsoft/vscode-dotnettools/issues/156), this is now supported.

However I think it could be useful to have the option to pass in some inline parameters to be used when executing the tests, without having to worry about creating and maintaining quite a complex XML configuration file for something that's so trivial.

**Comment by ChristoWolf** [comment]
Yes, "manual" coverage collection (e.g. via executing `dotnet test` with appropriate coverage collection settings/arguments) works of course, it's just that proper integration in VS Code is missing.

There seems to be no possibility to collect coverage using the C# Dev Kit Test Explorer, nor any coverage gutter visualizations.

**Comment by ChristoWolf** [comment]
See also #1035.

**Comment by sbward** [comment]
> I think it should be possible to use a [`.runsettings`](https://learn.microsoft.com/en-us/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file?view=vs-2022) with the relevant configuration to collect code coverage. As per the [linked issue](https://github.com/microsoft/vscode-dotnettools/issues/156), this is now supported.
> 
> However I think it could be useful to have the option to pass in some inline parameters to be used when executing the tests, without having to worry about creating and maintaining quite a complex XML configuration file for something that's so trivial.

I've just tried this, and although passing a `.runsettings` file through the config is possible, it is not a solution for the following reasons:

- Users might have multiple unit test projects with their own `.runsettings` that each produce their own `{Project}/TestResults/{GUID}` output directories which then need to be manually merged.
- While it is possible to use a single `.runsettings` in the root of a solution and reference it in the C# dev kit settings, when I tried this, the resulting `TestResults/{GUID}` folder was empty for some reason. I only had this issue when `.runsettings` was in the root of the solution (when I put `.runsettings` into a unit test project and referenced it there, I got coverage files as expected).
- Users currently need to install third-party coverage tools like Coverage Gutters or Code Coverage and then write VSCode Tasks that manually feed coverage files into these tools. This is difficult to get right.

**The permanent solution is to update C# Dev Kit to use the native coverage API for the Test Explorer panel.**

**Comment by dgtvan** [comment]
`...This is difficult to get right.` You're right, sir.
It would be much appreciated if code coverage becomes native, no double about it.

**Comment by ianjirka** [comment]
> * While it is possible to use a single `.runsettings` in the root of a solution and reference it in the C# dev kit settings, when I tried this, the resulting `TestResults/{GUID}` folder was empty for some reason. I only had this issue when `.runsettings` was in the root of the solution (when I put `.runsettings` into a unit test project and referenced it there, I got coverage files as expected).

I have this same issue.  When I run 'dotnet test --settings [path_to_runsettings]' it produces code coverage correctly but when I use C# devkit to run the unit tests, the {GUID} folder created but empty.
